components/pluralrules/src/operands.rs

```diff
@@ -200,3 +222,59 @@ macro_rules! impl_signed_integer_type {
 
 impl_integer_type!(u8 u16 u32 u64 u128 usize);
 impl_signed_integer_type!(i8 i16 i32 i64 i128 isize);
+
+impl From<&FixedDecimal> for PluralOperands {
+    /// Converts a [icu_num_util::FixedDecimal] to [PluralOperands]
+    fn from(f: &FixedDecimal) -> Self {
+        let mag_range = f.magnitude_range();
+        let mag_end = *mag_range.end();
+        let integer_range = 0..=mag_end;
+        let mag_start = *mag_range.start();
+
+        let mut integer_part: u64 = 0;
+        for magnitude in integer_range.rev() {
+            integer_part *= 10;
+            integer_part += f.digit_at(magnitude) as u64;
+        }
+
+        // The fractional part of the number, expressed as an integer to preserve trailing zeroes.
+        // For example, could be "100" if the stored number is "0.100".
+        let mut fraction_part_full: u64 = 0;
+        // A running total of the number of trailing zeros seen in the fractional part of the
+        // number.
+        let mut num_trailing_zeros = 0;
+        let mut num_fractional_digits = 0;
+        // 10^num_fractional_digits
+        let mut weight = 1;
+        // 10^num_trailing_zeros.
+        let mut weight_trailing_zeros = 1;
+        let fractional_magnitude_range = mag_start..=-1;
+        for magnitude in fractional_magnitude_range.rev() {
+            num_fractional_digits += 1;
```

Comment:
`num_fractional_digits` is just the length of `fractional_magnitude_range`.

Reply:
Yes, except the range doesn't have a length.  There is a way to compute it though, added.